drivers/sensor/lis2dw12/lis2dw12_spi.c

```diff
@@ -0,0 +1,180 @@
+/* ST Microelectronics LIS2DW12 3-axis accelerometer driver
+ *
+ * Copyright (c) 2019 STMicroelectronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Datasheet:
+ * https://www.st.com/resource/en/datasheet/lis2dw12.pdf
+ */
+
+
+#include <string.h>
+#include <spi.h>
+#include "lis2dw12.h"
+#include <logging/log.h>
+
+#ifdef DT_ST_LIS2DW12_BUS_SPI
+
+#define LIS2DW12_SPI_READ		(1 << 7)
+
+#define LOG_LEVEL CONFIG_SENSOR_LOG_LEVEL
+LOG_MODULE_DECLARE(LIS2DW12);
+
+#if defined(DT_ST_LIS2DW12_0_CS_GPIO_CONTROLLER)
+static struct spi_cs_control lis2dw12_cs_ctrl;
```

Comment:
Why aren't you using the .cs field in struct spi_config?

Reply:
Actually this is the driver istance of the spi_cs_control. Its address goes in .cs and is used by 
SPI driver.
If DT_ST_LIS2DW12_0_CS_GPIO_CONTROLLER is not defined then .cs remains NULL
and the SPI chip select will be generated by the controller itself.

Reply:
My mistake, I missed that this was .cs was a pointer. That said, why isn't the instance part of `struct lis2dw12_data`?

Reply:
OK, I can put it in data structure. It seems cleaner to me.